src/otx/algo/detection/d_fine.py

```diff
@@ -0,0 +1,312 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""D-Fine model implementations."""
+
+from __future__ import annotations
+
+import copy
+import re
+from typing import TYPE_CHECKING, Any, Literal
+
+import torch
+from torch import Tensor, nn
+from torchvision.ops import box_convert
+from torchvision.tv_tensors import BoundingBoxFormat
+
+from otx.algo.detection.backbones.hgnetv2 import HGNetv2
+from otx.algo.detection.detectors import DETR
+from otx.algo.detection.heads.dfine_decoder import DFINETransformer
+from otx.algo.detection.losses.dfine_loss import DFINECriterion
+from otx.algo.detection.necks.dfine_hybrid_encoder import HybridEncoder
+from otx.algo.utils.mmengine_utils import load_checkpoint
+from otx.core.config.data import TileConfig
+from otx.core.data.entity.base import OTXBatchLossEntity
+from otx.core.data.entity.detection import DetBatchDataEntity, DetBatchPredEntity
+from otx.core.exporter.base import OTXModelExporter
+from otx.core.exporter.native import OTXNativeModelExporter
+from otx.core.metrics.fmeasure import MeanAveragePrecisionFMeasureCallable
+from otx.core.model.base import DefaultOptimizerCallable, DefaultSchedulerCallable
+from otx.core.model.detection import ExplainableOTXDetModel
+
+if TYPE_CHECKING:
+    from lightning.pytorch.cli import LRSchedulerCallable, OptimizerCallable
+
+    from otx.core.metrics import MetricCallable
+    from otx.core.schedulers import LRSchedulerListCallable
+    from otx.core.types.label import LabelInfoTypes
+
+
+PRETRAINED_ROOT: str = "https://github.com/Peterande/storage/releases/download/dfinev1.0/"
+
+PRETRAINED_WEIGHTS: dict[str, str] = {
```

Comment:
I wonder whether we need all of these variants? We are currently overwhelmed with detection recipes. Could we choose maybe 2 models to expose and omit others? The largest one shows the best performance and it is a candidate for Geti largest template revamp, but other templates seems to be not so beneficial comparing with already introduced models.
So, I would consider cleaning some model versions here (same concerns RTDetr and YOLOX, but it is another story)